core/jobs/types/model_property.py

```diff
@@ -109,62 +128,95 @@ def yield_value_from_model(self, model):
         else:
             yield value
 
-    def _to_model_class(self):
+    def _to_model_class(self) -> Type[base_models.BaseModel]:
         """Returns the model class associated with this instance.
 
         Returns:
             type(BaseModel). The model type.
         """
-        return job_utils.get_model_class(self._model_kind)
-
-    def _to_property(self):
+        # Here, get_model_class() returns the value of type
+        # Type[datastore_services.Model] but the expected return value from
+        # this method is Type[base_models.BaseModel]. So, that's why we used
+        # cast here.
+        return cast(
+            Type[base_models.BaseModel],
+            job_utils.get_model_class(self._model_kind)
+        )
```

Comment:
Can we rather use `assert isinstance` for this?

Reply:
Done!